node/external/blockAPI/shardBlock.go

```diff
@@ -63,6 +74,22 @@ func (sbp *shardAPIBlockProcessor) GetBlockByHash(hash []byte, options api.Block
 		return nil, err
 	}
 
+	blockHeader, err := process.UnmarshalShardHeader(sbp.marshalizer, blockBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	// if genesis block, get the altered block bytes
+	if blockHeader.GetRound() == 0 {
+		alteredHash := make([]byte, 0)
```

Comment:
this is used in two places, shard and meta, can you extract the hash creation into a method?

Reply:
done